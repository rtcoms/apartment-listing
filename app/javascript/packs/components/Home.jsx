import React from 'react'
import PropTypes from 'prop-types'
import { Link } from "react-router-dom";
import ApartmentsFilterForm from './ApartmentsFilterForm.jsx';

const Home = props => (
  <React.Fragment>
    <div>Home {props.name}!</div>
    <Link to={`/apartments`}>View Apartments</Link>
  </React.Fragment>
)

Home.defaultProps = {
  name: 'David'
}

Home.propTypes = {
  name: PropTypes.string
}

export default Home;
