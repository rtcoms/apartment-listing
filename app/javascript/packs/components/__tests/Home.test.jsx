import Home from '../Home.jsx'

import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from "react-router-dom";

test('Home should render with name', () => {
  const component = renderer.create(
    <BrowserRouter><Home></Home></BrowserRouter>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
