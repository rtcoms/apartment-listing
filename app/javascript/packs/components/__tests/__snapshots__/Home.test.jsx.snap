// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render with name 1`] = `
<div
  className="chakra-stack css-1u3fgmi"
>
  <div
    className="css-45tizf"
  >
    <div
      className="chakra-stack css-n21gh5"
    >
      <h2
        className="chakra-heading css-0"
      >
        <span
          className="chakra-text css-79elbk"
        >
          Hello Kasaz
        </span>
        <br />
        <span
          className="chakra-text css-p03q1r"
        >
          Aparmtnet Listing
        </span>
      </h2>
      <p
        className="chakra-text css-mentm3"
      >
        This is an apartment listing management project
      </p>
      <div
        className="chakra-stack css-qz7axs"
      >
        <button
          className="chakra-button css-m56eyr"
          type="button"
        >
          <a
            href="/apartments"
            onClick={[Function]}
          >
            View Apartments
          </a>
        </button>
      </div>
    </div>
  </div>
</div>
`;
